backend/src/main/java/com/sep6/backend/service/AuthenticationService.java

```diff
@@ -0,0 +1,125 @@
+import com.sep6.backend.models.Account;
+import com.sep6.backend.models.auth.AuthenticationRequest;
+import com.sep6.backend.models.auth.AuthenticationResponse;
+import com.sep6.backend.models.auth.RegisterRequest;
+import com.sep6.backend.repository.AccountsRepository;
+import com.sep6.backend.repository.TokenRepository;
+import com.sep6.backend.security.config.JwtService;
+import com.sep6.backend.security.token.Token;
+import com.sep6.backend.security.token.TokenType;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.sql.Date;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationService {
+    private final AccountsRepository repository;
+    private final TokenRepository tokenJpaRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final JwtService jwtService;
+    private final AuthenticationManager authenticationManager;
+
+    public AuthenticationResponse register(RegisterRequest request) {
+        var user = Account.builder()
+                          .name(request.getName())
+                          .username(request.getUsername())
+                          .password(passwordEncoder.encode(request.getPassword()))
+                          .email(request.getEmail())
+                          .country(request.getCountry())
+                          .profilePictureUrl(request.getProfilePictureUrl())
+                          .dateOfBirth(Date.valueOf(request.getDateOfBirth()))
+                          .gender(request.getGender())
+                          .role(request.getRole())
+                          .build();
+        var savedUser = repository.save(user);
+        var jwtToken = jwtService.generateToken(user);
+        var refreshToken = jwtService.generateRefreshToken(user);
+        saveUserToken(savedUser, jwtToken);
+        return AuthenticationResponse.builder()
+                                     .accessToken(jwtToken)
+                                     .refreshToken(refreshToken)
+                                     .build();
+    }
+
+    public AuthenticationResponse authenticate(AuthenticationRequest request) {
+        authenticationManager.authenticate(
+                new UsernamePasswordAuthenticationToken(
+                        request.getEmail(),
+                        request.getPassword()
+                )
+        );
+        var user = repository.findByEmail(request.getEmail())
+                             .orElseThrow();
+        var jwtToken = jwtService.generateToken(user);
```

Comment:
We should check if the user is valid or not, to make sure that the deleted users cannot login

Reply:
This can be done in a new PR or here, but I saw that there is a method in Account called isAccountNonExpired created with a dummy var.

Reply:
we create a separate issue for that